pkg/cmd/create/create_devpod.go

```diff
@@ -158,6 +158,20 @@ func NewCmdCreateDevPod(commonOpts *opts.CommonOptions) *cobra.Command {
 	return cmd
 }
 
+// GetGitLabel try to reduce the label key and value size if they exceed 64 characters in length
+func GetGitLabel(gitInfo *gits.GitRepository, importURL string) (string, string) {
+	gitLabelKey := fmt.Sprintf("%s-%s-%s-%s", kube.LabelDevPodGitPrefix, gitInfo.Host, gitInfo.Organisation, gitInfo.Name)
+	gitLabelValue := naming.ToValidNameWithDots(importURL)
+
+	// try to minimize label size reducing git host to first part
+	if len(gitLabelKey) > 63 || len(gitLabelValue) > 63 {
+		gitLabelKey = fmt.Sprintf("%s-%s-%s-%s", kube.LabelDevPodGitPrefix, gitInfo.Host[0:strings.Index(gitInfo.Host, ".")], gitInfo.Organisation, gitInfo.Name)
```

Comment:
Does it matter how the label key is truncated? What if even with the host part the length is exceeded? I guess by just truncating you might get potential clashes in label selection, right?

One wonders why this all needs to go into a single label? Would it not be better to have three separate labels, host, organisation and name? Each of which should be wrapped around a `ToValidNameTruncated`. WDYT?

Reply:
You are right, this patch don't remove the problem of length, it only tries to reduce its possible occurrence.

And you are right again .. I agree with you, we must separate the host, the group and the project name in different labels, but not only at this point, in all the places where they are used to create labels.

This little patch is a quick win to reduce the ocurrence, but don't fix the problem

Reply:
> You are right, this patch don't remove the problem of length, it only tries to reduce its possible occurrence.

Ok

> And you are right again .. I agree with you, we must separate the host, the group and the project name in different labels, but not only at this point, in all the places where they are used to create labels.

These labels are just used within the devpods, right? So we could just cleanup the devpod code to use individual labels, right? Would that not be the proper fix?

Reply:
Ok for me. I'm workking on it !

Reply:
Hi,
  four labels for scheme, host, org and repo is ok for you ? For example

jenkins.io/repo/scheme: https
jenkins.io/repo/host: git.very-large-name-of-my-company.com
jenkins.io/repo/organisation: large-name-group
jenkins.io/repo/name: project-1

j